packages/clarity-js/src/layout/mutation.ts

```diff
@@ -145,6 +150,42 @@ async function process(): Promise<void> {
     task.stop(timer);
 }
 
+function track(m: MutationRecord, timer: Metric): string {
+  let value = m.target ? dom.get(m.target.parentNode) : null;
+  if (value) {
+    let inactive = time() > activePeriod;
+    // We use selector, instead of id, to determine the key because in some cases
+    // repeated mutations can cause elements to be destroyed and then recreated as new DOM nodes
+    // In those cases, IDs will change however the selector (which is relative to DOM xPath) remains the same
+    let key = [value.selector, m.attributeName, m.addedNodes ? m.addedNodes.length : 0, m.removedNodes ? m.removedNodes.length : 0].join();
```

Comment:
ARe there any cases they might cause issues? Just applying "how can this break" thinking.

Reply:
For example a false positive or false negative based on same key when it should be a different key (or vice versa)?

Reply:
We discussed this offline. It errs on the side of letting the mutation pass through if it's not the exactly same signature.